utils/getCookie.ts

```diff
@@ -0,0 +1,15 @@
+import { GetServerSidePropsContext } from "next";
+
+export const getCookie = (name: string) => {
+  if (typeof window !== "undefined") {
+    let matches = document.cookie.match(new RegExp("(?:^|; )" + name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, "\\$1") + "=([^;]*)"));
+    return matches ? decodeURIComponent(matches[1]) : undefined;
+  }
+};
+
+export const getServerCookie = (context: GetServerSidePropsContext, name: string) => {
+  if (context.req.headers.cookie) {
+    let matches = context.req.headers.cookie.match(new RegExp("(?:^|; )" + name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, "\\$1") + "=([^;]*)"));
+    return matches ? decodeURIComponent(matches[1]) : undefined;
+  }
+};
```

Comment:
정규식이 중복된 것 같아서, 다음 코드처럼 재사용 함수를 사용해봐도 좋을 것 같습니다:
```suggestion
const extractCookieValue = (cookieString: string, name: string) => {
  let matches = cookieString.match(new RegExp("(?:^|; )" + name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, "\\$1") + "=([^;]*)"));
  return matches ? decodeURIComponent(matches[1]) : undefined;
};

export const getCookie = (name: string) => {
  if (typeof window !== "undefined") {
    return extractCookieValue(document.cookie, name);
  }
};

export const getServerCookie = (context: GetServerSidePropsContext, name: string) => {
  const cookieString = context.req.headers.cookie;
  if (cookieString) {
    return extractCookieValue(cookieString, name);
  }
};

```